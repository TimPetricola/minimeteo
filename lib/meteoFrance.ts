import { number } from "fp-ts";
import { isLeft } from "fp-ts/lib/Either";
import * as t from "io-ts";
import { PathReporter } from "io-ts/lib/PathReporter";
import { HourlyForecast } from "./types";

const API_BASE_URL = "https://webservice.meteofrance.com";
const METEO_FRANCE_TOKEN = "__Wj7dVSTjV9YGu1guveLyDq0g7S7TfTjaHBTPTpO0kj8__";
const LANG = "en";

const request = async <T>(
  path: string,
  params: Record<string, string>
): Promise<T> => {
  const searchParams = new URLSearchParams({
    ...params,
    lang: LANG,
    token: METEO_FRANCE_TOKEN,
  });
  const response = await fetch(
    `${API_BASE_URL}${path}?${searchParams.toString()}`
  );

  if (!response.ok) throw new Error("Network response was not ok");

  return response.json();
};

export const searchPlaces = async (query: string) => {
  const places = await request<
    {
      name: string;
      lat: number;
      lon: number;
      country: number;
    }[]
  >("/places", { q: query });

  return {
    places: places.map((place) => ({
      name: place.name,
      lat: place.lat,
      lon: place.lon,
      country: place.country,
    })),
  };
};

export const parseResponse = <TData extends any = any, O = TData, I = unknown>(
  rawData: I,
  codec: t.Type<TData, O, I>
): TData => {
  const result = codec.decode(rawData);
  if (isLeft(result)) throw new Error(PathReporter.report(result).join("\n"));

  return result.right;
};

const RawPosition = t.type({
  lat: t.number,
  lon: t.number,
  alti: t.number,
  name: t.string,
  timezone: t.string,
  rain_product_available: t.union([t.literal(0), t.literal(1)]),
});

const RawForecast = t.type({
  position: RawPosition,
  daily_forecast: t.array(
    t.type({
      dt: t.number,
      T: t.type({
        min: t.number,
        max: t.number,
      }),
    })
  ),
  forecast: t.array(
    t.type({
      dt: t.number,
      T: t.type({
        value: t.number,
        windchill: t.number,
      }),
      weather: t.type({
        icon: t.string,
        desc: t.string,
      }),
    })
  ),
});

const RawRainForecast = t.type({
  forecast: t.array(
    t.type({
      dt: t.number,
      rain: t.number,
      desc: t.string,
    })
  ),
});

export const parseForecastResponse = (payload: any) =>
  parseResponse(payload, RawForecast);

export const parseRainForecastResponse = (payload: any) =>
  parseResponse(payload, RawRainForecast);

export const sanitizeForecast = (
  payload: t.TypeOf<typeof RawForecast>
): {
  position: {
    altitude: number;
    latitude: number;
    longitude: number;
    name: string;
    timeZone: string;
    isRainForecastAvailable: boolean;
  };
  hourly: HourlyForecast[];
  daily: {
    datetime: Date;
    temperature: { min: number; max: number };
  }[];
} => {
  return {
    position: {
      altitude: payload.position.alti,
      latitude: payload.position.lat,
      longitude: payload.position.lon,
      name: payload.position.name,
      timeZone: payload.position.timezone,
      isRainForecastAvailable: payload.position.rain_product_available === 1,
    },
    hourly: payload.forecast
      .filter(
        (raw) =>
          typeof raw.T.windchill === "number" &&
          typeof raw.weather.icon === "string" &&
          typeof raw.weather.desc === "string"
      )
      .map((raw) => ({
        datetime: new Date(raw.dt * 1000),
        temperature: raw.T.value,
        perceivedTemperature: raw.T.windchill,
        weatherDescription: raw.weather.desc,
        iconId: raw.weather.icon,
      })),
    daily: payload.daily_forecast
      .filter(
        (raw) => typeof raw.T.min === "number" && typeof raw.T.max === "number"
      )
      .map((raw) => ({
        datetime: new Date(raw.dt * 1000),
        temperature: { min: raw.T.min, max: raw.T.max },
      })),
  };
};

export const sanitizeRainForecast = (
  payload: t.TypeOf<typeof RawRainForecast>
): {
  upcomingRain: { datetime: Date; value: number; description: string }[];
} => {
  return {
    upcomingRain: payload.forecast.map((raw) => ({
      datetime: new Date(raw.dt * 1000),
      value: raw.rain,
      description: raw.desc,
    })),
  };
};

export const fetchForecast = async (
  lat: string | number,
  lon: string | number
) => {
  const payload = await request("/forecast", {
    lat: lat.toString(),
    lon: lon.toString(),
  });

  return sanitizeForecast(parseForecastResponse(payload));
};

export const fetchRainForecast = async (
  lat: string | number,
  lon: string | number
) => {
  const payload = await request(`/rain`, {
    lat: lat.toString(),
    lon: lon.toString(),
  });

  return sanitizeRainForecast(parseRainForecastResponse(payload));
};
