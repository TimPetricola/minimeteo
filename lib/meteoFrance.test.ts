import { parseForecastResponse, sanitizeForecast } from "./meteoFrance";

describe(".parseForecastResponse", () => {
  it("does nothing when valid", () => {
    expect(() =>
      parseForecastResponse(
        require("./__tests__/payloads/forecast/annecy-2021-09-24-09-14.json")
      )
    ).not.toThrow();
  });

  it("throws when invalid", () => {
    expect(() =>
      parseForecastResponse({
        ...require("./__tests__/payloads/forecast/annecy-2021-09-24-09-14.json"),
        position: "foo",
      })
    ).toThrow(/Invalid value/);
  });
});

it(".sanitizeForecast", () => {
  const data = sanitizeForecast(
    require("./__tests__/payloads/forecast/annecy-2021-09-24-09-14.json")
  );

  expect(data.position).toMatchInlineSnapshot(`
    Object {
      "altitude": 447,
      "isRainForecastAvailable": false,
      "latitude": 45.898493,
      "longitude": 6.128991,
      "name": "Annecy",
      "timeZone": "Europe/Paris",
    }
  `);
  expect(data.hourly).toMatchSnapshot();
  expect(data.daily).toMatchSnapshot();
});
