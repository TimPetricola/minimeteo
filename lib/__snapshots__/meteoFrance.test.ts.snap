// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.sanitizeForecast 2`] = `
Array [
  Object {
    "datetime": 2021-09-24T06:00:00.000Z,
    "iconId": "p3j",
    "perceivedTemperature": 10.1,
    "temperature": 10.1,
    "weatherDescription": "Very cloudy",
  },
  Object {
    "datetime": 2021-09-24T07:00:00.000Z,
    "iconId": "p2j",
    "perceivedTemperature": 13,
    "temperature": 13,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-24T08:00:00.000Z,
    "iconId": "p2j",
    "perceivedTemperature": 17.3,
    "temperature": 15.9,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-24T09:00:00.000Z,
    "iconId": "p2j",
    "perceivedTemperature": 20.6,
    "temperature": 18.8,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-24T10:00:00.000Z,
    "iconId": "p1j",
    "perceivedTemperature": 22.1,
    "temperature": 20,
    "weatherDescription": "Sunny",
  },
  Object {
    "datetime": 2021-09-24T11:00:00.000Z,
    "iconId": "p1j",
    "perceivedTemperature": 23.5,
    "temperature": 21.2,
    "weatherDescription": "Sunny",
  },
  Object {
    "datetime": 2021-09-24T12:00:00.000Z,
    "iconId": "p1j",
    "perceivedTemperature": 25,
    "temperature": 22.4,
    "weatherDescription": "Sunny",
  },
  Object {
    "datetime": 2021-09-24T13:00:00.000Z,
    "iconId": "p1j",
    "perceivedTemperature": 25.3,
    "temperature": 22.8,
    "weatherDescription": "Sunny",
  },
  Object {
    "datetime": 2021-09-24T14:00:00.000Z,
    "iconId": "p1j",
    "perceivedTemperature": 25.5,
    "temperature": 23.3,
    "weatherDescription": "Sunny",
  },
  Object {
    "datetime": 2021-09-24T15:00:00.000Z,
    "iconId": "p1j",
    "perceivedTemperature": 25.8,
    "temperature": 23.7,
    "weatherDescription": "Sunny",
  },
  Object {
    "datetime": 2021-09-24T16:00:00.000Z,
    "iconId": "p1j",
    "perceivedTemperature": 23.9,
    "temperature": 22,
    "weatherDescription": "Sunny",
  },
  Object {
    "datetime": 2021-09-24T17:00:00.000Z,
    "iconId": "p1j",
    "perceivedTemperature": 22,
    "temperature": 20.3,
    "weatherDescription": "Sunny",
  },
  Object {
    "datetime": 2021-09-24T18:00:00.000Z,
    "iconId": "p1j",
    "perceivedTemperature": 20.1,
    "temperature": 18.5,
    "weatherDescription": "Sunny",
  },
  Object {
    "datetime": 2021-09-24T19:00:00.000Z,
    "iconId": "p1n",
    "perceivedTemperature": 18.7,
    "temperature": 17.2,
    "weatherDescription": "Clear night",
  },
  Object {
    "datetime": 2021-09-24T20:00:00.000Z,
    "iconId": "p1n",
    "perceivedTemperature": 17.2,
    "temperature": 15.9,
    "weatherDescription": "Clear night",
  },
  Object {
    "datetime": 2021-09-24T21:00:00.000Z,
    "iconId": "p1n",
    "perceivedTemperature": 14.5,
    "temperature": 14.5,
    "weatherDescription": "Clear night",
  },
  Object {
    "datetime": 2021-09-24T22:00:00.000Z,
    "iconId": "p1n",
    "perceivedTemperature": 14.1,
    "temperature": 14.1,
    "weatherDescription": "Clear night",
  },
  Object {
    "datetime": 2021-09-24T23:00:00.000Z,
    "iconId": "p1n",
    "perceivedTemperature": 13.6,
    "temperature": 13.6,
    "weatherDescription": "Clear night",
  },
  Object {
    "datetime": 2021-09-25T00:00:00.000Z,
    "iconId": "p1n",
    "perceivedTemperature": 13.1,
    "temperature": 13.1,
    "weatherDescription": "Clear night",
  },
  Object {
    "datetime": 2021-09-25T01:00:00.000Z,
    "iconId": "p1n",
    "perceivedTemperature": 12.7,
    "temperature": 12.7,
    "weatherDescription": "Clear night",
  },
  Object {
    "datetime": 2021-09-25T02:00:00.000Z,
    "iconId": "p1n",
    "perceivedTemperature": 12.4,
    "temperature": 12.4,
    "weatherDescription": "Clear night",
  },
  Object {
    "datetime": 2021-09-25T03:00:00.000Z,
    "iconId": "p1n",
    "perceivedTemperature": 11.8,
    "temperature": 12.1,
    "weatherDescription": "Clear night",
  },
  Object {
    "datetime": 2021-09-25T04:00:00.000Z,
    "iconId": "p2n",
    "perceivedTemperature": 11.9,
    "temperature": 12.1,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-25T05:00:00.000Z,
    "iconId": "p2n",
    "perceivedTemperature": 12.1,
    "temperature": 12.2,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-25T06:00:00.000Z,
    "iconId": "p2j",
    "perceivedTemperature": 12.3,
    "temperature": 12.3,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-25T07:00:00.000Z,
    "iconId": "p1j",
    "perceivedTemperature": 14.8,
    "temperature": 14.8,
    "weatherDescription": "Sunny",
  },
  Object {
    "datetime": 2021-09-25T08:00:00.000Z,
    "iconId": "p1j",
    "perceivedTemperature": 20,
    "temperature": 17.4,
    "weatherDescription": "Sunny",
  },
  Object {
    "datetime": 2021-09-25T09:00:00.000Z,
    "iconId": "p1j",
    "perceivedTemperature": 22.9,
    "temperature": 20,
    "weatherDescription": "Sunny",
  },
  Object {
    "datetime": 2021-09-25T10:00:00.000Z,
    "iconId": "p2j",
    "perceivedTemperature": 24.7,
    "temperature": 21.4,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-25T11:00:00.000Z,
    "iconId": "p2j",
    "perceivedTemperature": 26.6,
    "temperature": 22.9,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-25T12:00:00.000Z,
    "iconId": "p2j",
    "perceivedTemperature": 28.4,
    "temperature": 24.3,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-25T13:00:00.000Z,
    "iconId": "p2j",
    "perceivedTemperature": 29.1,
    "temperature": 24.6,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-25T14:00:00.000Z,
    "iconId": "p2j",
    "perceivedTemperature": 29.9,
    "temperature": 24.8,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-25T15:00:00.000Z,
    "iconId": "p2j",
    "perceivedTemperature": 30.6,
    "temperature": 25.1,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-25T16:00:00.000Z,
    "iconId": "p2j",
    "perceivedTemperature": 29.5,
    "temperature": 23.6,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-25T17:00:00.000Z,
    "iconId": "p2j",
    "perceivedTemperature": 28.2,
    "temperature": 22.2,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-25T18:00:00.000Z,
    "iconId": "p2n",
    "perceivedTemperature": 26.8,
    "temperature": 20.7,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-25T19:00:00.000Z,
    "iconId": "p26n",
    "perceivedTemperature": 25.3,
    "temperature": 19.6,
    "weatherDescription": "Risk of storms",
  },
  Object {
    "datetime": 2021-09-25T20:00:00.000Z,
    "iconId": "p26n",
    "perceivedTemperature": 23.7,
    "temperature": 18.6,
    "weatherDescription": "Risk of storms",
  },
  Object {
    "datetime": 2021-09-25T21:00:00.000Z,
    "iconId": "p26n",
    "perceivedTemperature": 22.2,
    "temperature": 17.5,
    "weatherDescription": "Risk of storms",
  },
  Object {
    "datetime": 2021-09-25T22:00:00.000Z,
    "iconId": "p26n",
    "perceivedTemperature": 21.5,
    "temperature": 17.1,
    "weatherDescription": "Risk of storms",
  },
  Object {
    "datetime": 2021-09-25T23:00:00.000Z,
    "iconId": "p26n",
    "perceivedTemperature": 20.9,
    "temperature": 16.7,
    "weatherDescription": "Risk of storms",
  },
  Object {
    "datetime": 2021-09-26T00:00:00.000Z,
    "iconId": "p26n",
    "perceivedTemperature": 20.2,
    "temperature": 16.3,
    "weatherDescription": "Risk of storms",
  },
  Object {
    "datetime": 2021-09-26T03:00:00.000Z,
    "iconId": "p13bisn",
    "perceivedTemperature": 18.2,
    "temperature": 15.3,
    "weatherDescription": "Few showers",
  },
  Object {
    "datetime": 2021-09-26T06:00:00.000Z,
    "iconId": "p13bisn",
    "perceivedTemperature": 19.6,
    "temperature": 15.8,
    "weatherDescription": "Few showers",
  },
  Object {
    "datetime": 2021-09-26T09:00:00.000Z,
    "iconId": "p16j",
    "perceivedTemperature": 23.9,
    "temperature": 18.6,
    "weatherDescription": "Stormy rain",
  },
  Object {
    "datetime": 2021-09-26T12:00:00.000Z,
    "iconId": "p16j",
    "perceivedTemperature": 25.1,
    "temperature": 19.3,
    "weatherDescription": "Stormy rain",
  },
  Object {
    "datetime": 2021-09-26T15:00:00.000Z,
    "iconId": "p14bisj",
    "perceivedTemperature": 25.2,
    "temperature": 20.1,
    "weatherDescription": "Showers",
  },
  Object {
    "datetime": 2021-09-26T18:00:00.000Z,
    "iconId": "p14bisj",
    "perceivedTemperature": 21.7,
    "temperature": 17.4,
    "weatherDescription": "Showers",
  },
  Object {
    "datetime": 2021-09-26T21:00:00.000Z,
    "iconId": "p12bisn",
    "perceivedTemperature": 14.8,
    "temperature": 14.8,
    "weatherDescription": "Few showers",
  },
  Object {
    "datetime": 2021-09-27T00:00:00.000Z,
    "iconId": "p12bisn",
    "perceivedTemperature": 13.8,
    "temperature": 13.6,
    "weatherDescription": "Few showers",
  },
  Object {
    "datetime": 2021-09-27T03:00:00.000Z,
    "iconId": "p4n",
    "perceivedTemperature": 13,
    "temperature": 13,
    "weatherDescription": "Hazy sky",
  },
  Object {
    "datetime": 2021-09-27T06:00:00.000Z,
    "iconId": "p4n",
    "perceivedTemperature": 12.3,
    "temperature": 12.4,
    "weatherDescription": "Hazy sky",
  },
  Object {
    "datetime": 2021-09-27T09:00:00.000Z,
    "iconId": "p4j",
    "perceivedTemperature": 20.1,
    "temperature": 16.7,
    "weatherDescription": "Hazy sky",
  },
  Object {
    "datetime": 2021-09-27T12:00:00.000Z,
    "iconId": "p4j",
    "perceivedTemperature": 25.2,
    "temperature": 20.9,
    "weatherDescription": "Hazy sky",
  },
  Object {
    "datetime": 2021-09-27T15:00:00.000Z,
    "iconId": "p4j",
    "perceivedTemperature": 23.4,
    "temperature": 19.5,
    "weatherDescription": "Hazy sky",
  },
  Object {
    "datetime": 2021-09-27T18:00:00.000Z,
    "iconId": "p4j",
    "perceivedTemperature": 21.5,
    "temperature": 18.2,
    "weatherDescription": "Hazy sky",
  },
  Object {
    "datetime": 2021-09-28T00:00:00.000Z,
    "iconId": "p12n",
    "perceivedTemperature": 13.6,
    "temperature": 13.6,
    "weatherDescription": "Scattered rain",
  },
  Object {
    "datetime": 2021-09-28T06:00:00.000Z,
    "iconId": "p12n",
    "perceivedTemperature": 11.8,
    "temperature": 11.8,
    "weatherDescription": "Scattered rain",
  },
  Object {
    "datetime": 2021-09-28T12:00:00.000Z,
    "iconId": "p2j",
    "perceivedTemperature": 18.8,
    "temperature": 18.8,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-28T18:00:00.000Z,
    "iconId": "p12bisj",
    "perceivedTemperature": 15.9,
    "temperature": 15.9,
    "weatherDescription": "Few showers",
  },
  Object {
    "datetime": 2021-09-29T00:00:00.000Z,
    "iconId": "p12bisn",
    "perceivedTemperature": 12.3,
    "temperature": 12.3,
    "weatherDescription": "Few showers",
  },
  Object {
    "datetime": 2021-09-29T06:00:00.000Z,
    "iconId": "p14bisn",
    "perceivedTemperature": 11.4,
    "temperature": 11.4,
    "weatherDescription": "Showers",
  },
  Object {
    "datetime": 2021-09-29T12:00:00.000Z,
    "iconId": "p14bisj",
    "perceivedTemperature": 17.2,
    "temperature": 17.2,
    "weatherDescription": "Showers",
  },
  Object {
    "datetime": 2021-09-29T18:00:00.000Z,
    "iconId": "p14bisj",
    "perceivedTemperature": 14.4,
    "temperature": 14.4,
    "weatherDescription": "Showers",
  },
  Object {
    "datetime": 2021-09-30T00:00:00.000Z,
    "iconId": "p14bisn",
    "perceivedTemperature": 11.1,
    "temperature": 11.1,
    "weatherDescription": "Showers",
  },
  Object {
    "datetime": 2021-09-30T06:00:00.000Z,
    "iconId": "p2n",
    "perceivedTemperature": 9.3,
    "temperature": 9.3,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-30T12:00:00.000Z,
    "iconId": "p2j",
    "perceivedTemperature": 15.01,
    "temperature": 15,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-30T18:00:00.000Z,
    "iconId": "p2j",
    "perceivedTemperature": 12.9,
    "temperature": 12.9,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-10-01T00:00:00.000Z,
    "iconId": "p2n",
    "perceivedTemperature": 9.9,
    "temperature": 9.9,
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-10-01T06:00:00.000Z,
    "iconId": "p14bisn",
    "perceivedTemperature": 8.4,
    "temperature": 8.4,
    "weatherDescription": "Showers",
  },
  Object {
    "datetime": 2021-10-01T12:00:00.000Z,
    "iconId": "p14bisj",
    "perceivedTemperature": 15.8,
    "temperature": 15.8,
    "weatherDescription": "Showers",
  },
  Object {
    "datetime": 2021-10-01T18:00:00.000Z,
    "iconId": "p14bisj",
    "perceivedTemperature": 13.9,
    "temperature": 13.9,
    "weatherDescription": "Showers",
  },
  Object {
    "datetime": 2021-10-02T00:00:00.000Z,
    "iconId": "p14bisn",
    "perceivedTemperature": 10.4,
    "temperature": 10.4,
    "weatherDescription": "Showers",
  },
  Object {
    "datetime": 2021-10-02T06:00:00.000Z,
    "iconId": "p14bisn",
    "perceivedTemperature": 9,
    "temperature": 9,
    "weatherDescription": "Showers",
  },
  Object {
    "datetime": 2021-10-02T12:00:00.000Z,
    "iconId": "p14bisj",
    "perceivedTemperature": 15.5,
    "temperature": 15.5,
    "weatherDescription": "Showers",
  },
  Object {
    "datetime": 2021-10-02T18:00:00.000Z,
    "iconId": "p14bisj",
    "perceivedTemperature": 13.3,
    "temperature": 13.3,
    "weatherDescription": "Showers",
  },
  Object {
    "datetime": 2021-10-03T00:00:00.000Z,
    "iconId": "p14bisn",
    "perceivedTemperature": 10.1,
    "temperature": 10.1,
    "weatherDescription": "Showers",
  },
]
`;

exports[`.sanitizeForecast 3`] = `
Array [
  Object {
    "datetime": 2021-09-24T00:00:00.000Z,
    "iconId": "p1j",
    "temperature": Object {
      "max": 24.4,
      "min": 9.7,
    },
    "weatherDescription": "Sunny",
  },
  Object {
    "datetime": 2021-09-25T00:00:00.000Z,
    "iconId": "p2j",
    "temperature": Object {
      "max": 26,
      "min": 10.7,
    },
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-09-26T00:00:00.000Z,
    "iconId": "p16j",
    "temperature": Object {
      "max": 21.1,
      "min": 14,
    },
    "weatherDescription": "Stormy rain",
  },
  Object {
    "datetime": 2021-09-27T00:00:00.000Z,
    "iconId": "p4j",
    "temperature": Object {
      "max": 22.6,
      "min": 10.8,
    },
    "weatherDescription": "Hazy sky",
  },
  Object {
    "datetime": 2021-09-28T00:00:00.000Z,
    "iconId": "p12bisj",
    "temperature": Object {
      "max": 20.8,
      "min": 10.9,
    },
    "weatherDescription": "Few showers",
  },
  Object {
    "datetime": 2021-09-29T00:00:00.000Z,
    "iconId": "p14bisj",
    "temperature": Object {
      "max": 18.7,
      "min": 10.1,
    },
    "weatherDescription": "Showers",
  },
  Object {
    "datetime": 2021-09-30T00:00:00.000Z,
    "iconId": "p2j",
    "temperature": Object {
      "max": 16.5,
      "min": 8.2,
    },
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-10-01T00:00:00.000Z,
    "iconId": "p14bisj",
    "temperature": Object {
      "max": 17.1,
      "min": 7.3,
    },
    "weatherDescription": "Showers",
  },
  Object {
    "datetime": 2021-10-02T00:00:00.000Z,
    "iconId": "p14bisj",
    "temperature": Object {
      "max": 17,
      "min": 7.8,
    },
    "weatherDescription": "Showers",
  },
  Object {
    "datetime": 2021-10-03T00:00:00.000Z,
    "iconId": "p14j",
    "temperature": Object {
      "max": 18.1,
      "min": 7.7,
    },
    "weatherDescription": "Rain",
  },
  Object {
    "datetime": 2021-10-04T00:00:00.000Z,
    "iconId": "p2j",
    "temperature": Object {
      "max": 19.5,
      "min": 7.2,
    },
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-10-05T00:00:00.000Z,
    "iconId": "p1j",
    "temperature": Object {
      "max": 19.2,
      "min": 7.5,
    },
    "weatherDescription": "Sunny",
  },
  Object {
    "datetime": 2021-10-06T00:00:00.000Z,
    "iconId": "p2j",
    "temperature": Object {
      "max": 19.1,
      "min": 7.8,
    },
    "weatherDescription": "Cloudy",
  },
  Object {
    "datetime": 2021-10-07T00:00:00.000Z,
    "iconId": "p2j",
    "temperature": Object {
      "max": 18.7,
      "min": 7.3,
    },
    "weatherDescription": "Cloudy",
  },
]
`;
